ibis/backends/sqlite/__init__.py

```diff
@@ -41,4 +41,5 @@ def connect(self, path=None, create=False):
         create : boolean, default False
             If file does not exist, create it
         """
-        return SQLiteClient(backend=self, path=path, create=create)
+        self.client = SQLiteClient(backend=self, path=path, create=create)
```

Comment:
I realize this PR is old, but is it possible to clarify why a reference cycle is necessary here? Why do both the client and backend need references to each other?

Reply:
This is temporary, and was implemented to let transition from Client to Backend step by step. So, we had manageable PRs instead of a huge one.

The reason of the references is that any method moved from Client to Backend using self, had to use self.client instead. And methods not yet moved that called moved methods, had to use self.backend instead of the original self.

Will get rid of it soon, as I'm workingon merging the remaining methods in Client into Backend.

Does it make sense?

Reply:
Makes sense. Thanks for clarifying.